### ISSUE TITLE
Conflicting npm package name - hard fail

### ISSUE THREAD

**Issue by apowers313** [semantic-release/semantic-release#83]
I just created a [new project called `testing`](https://github.com/apowers313/testing), complete with the `package.json` declaring that the package is named `testing`. While that was foolish of me, since there is already an npm [package named `testing` that is owned by someone other than me](https://www.npmjs.com/package/testing), `semantic-release` doesn't catch the mistake and hard-fails instead:

```
semantic-release ERR! pre Failed to determine new version.
semantic-release ERR! pre 129 Command failed: /bin/sh -c git branch --contains 47c10285877c1bd241a7148ef11285439042f6f7
semantic-release ERR! pre error: no such commit 47c10285877c1bd241a7148ef11285439042f6f7
semantic-release ERR! pre usage: git branch [options] [-r | -a] [--merged | --no-merged]
semantic-release ERR! pre    or: git branch [options] [-l] [-f] <branchname> [<start-point>]
semantic-release ERR! pre    or: git branch [options] [-r] (-d | -D) <branchname>...
semantic-release ERR! pre    or: git branch [options] (-m | -M) [<oldbranch>] <newbranch>
[ ... etc ... ]
```

Maybe `@semantic_release/last-release-npm` should return `data.contributors` and `data.repository` in the objects passed to its callback so that `semantic_release`'s `pre.js` can check to see if `config.pkg.author` or `config.pkg.repository` match (respectively).

**Comment by boennemann** [comment]
Hey @apowers313,

thanks for reporting this. Turns out this just revealed a bug in the way the commits were extracted.
The commit hash not even being in the history is a source for a lot of confusion and problems, which is why I added the `git branch --contains` check in the first place. The bug is that I was assuming an empty stdout in this case, while in reality this also fails the command. The error shouldn't abort the flow, but be expected here: https://github.com/semantic-release/semantic-release/blob/caribou/src/lib/commits.js#L15

So assuming this bug wouldn't have existed when you tried this out the output would have been this:

> The commit the last release of this package was derived from is no longer
in the direct history of the "${branch}" branch.
This means semantic-release can not extract the commits between now and then.
This is usually caused by force pushing or releasing from an unrelated branch.
You can recover from this error by publishing manually or restoring
the commit "${from}".

I think this could've already been helpful, but we can easily add to the error message:
> This is usually caused by force pushing, releasing from an unrelated branch, or using an already existing package name.

 Given that the "last-release" part is now a plugin, we can no longer assume any information other than the commit-ish.

Do you think fixing the bug and modifying the error message is a sufficient way of dealing with this?

**Comment by apowers313** [comment]
Looks good to me, thanks for the consideration.

**Comment by boennemann** [comment]
The fix landed in 4.3.5 and 6.0.2 respectively :+1:
